compiler/src/dotty/tools/dotc/transform/localopt/Simplify.scala

```diff
@@ -146,4 +146,15 @@ object Simplify {
       case _ => None
     }
   }
+
+  // System.in is static final fields that, for legacy reasons, must be
```

Comment:
is -> is a
fields -> field

This description is a bit confusing: it starts by talking just about System.in and then mention the rest.
The name of the method is also not great, it's too easy to confuse with `is(Mutable)`, I would call it something like `isEffectivelyMutable`, `isReallyMutable` or `isChanging`